app/state_machines/flood_risk_engine/enrollment_state_machine.rb

```diff
@@ -14,8 +14,54 @@ class EnrollmentStateMachine < FiniteMachine::Definition
     initial :check_location
 
     events do
-      event :go_forward, WorkFlow.for(:local_authority)
-      event :go_back, WorkFlow.for(:local_authority).invert
+      event :go_forward, WorkFlow.for(:start).merge(
+        if: -> { target.org_type.nil? }
+      )
+      event :go_back, WorkFlow.for(:start).invert.merge(
+        if: -> { target.org_type.nil? }
+      )
+
+      event :go_forward, WorkFlow.for(:local_authority).merge(
+        if: -> { target.org_type == "local_authority" }
+      )
+      event :go_back, WorkFlow.for(:local_authority).invert.merge(
+        if: -> { target.org_type == "local_authority" }
+      )
+
+      event :go_forward, WorkFlow.for(:limited_company).merge(
+        if: -> { target.org_type == "limited_company" }
+      )
+      event :go_back, WorkFlow.for(:limited_company).invert.merge(
+        if: -> { target.org_type == "limited_company" }
+      )
+
+      event :go_forward, WorkFlow.for(:limited_liability_partnership).merge(
```

Comment:
couldn't this all go in some sort of loop, based on the complete list of org_type ?  .. make it much more readable and less brittle to change